frontend/src/components/accounts/AccountDetails.tsx

```diff
@@ -1,99 +1,146 @@
 
 import Balance from "../utils/Balance";
 import CardCell from "../utils/CardCell";
+import TransactionLink from "../utils/TransactionLink";
+import BlockLink from "../utils/BlockLink";
 
 export interface Props {
-  account: Account;
+  account: A.Account;
+}
+
+interface State {
+  storagePaidAtBlockHash: string | null;
 }
 
-export default ({ account }: Props) => {
-  return (
-    <div className="account-info-container">
-      <Row noGutters>
-        <Col md="3">
-          <CardCell
-            title="Ⓝ Balance"
-            text={<Balance amount={account.amount} />}
-            className="border-0"
-          />
-        </Col>
-        <Col md="3">
-          <CardCell
-            title="Transactions"
-            imgLink="/static/images/icon-m-transaction.svg"
-            text={
-              <>
-                <span>
-                  &uarr;{account.outTransactionsCount.toLocaleString()}
-                </span>
-                &nbsp;&nbsp;
-                <span>
-                  &darr;{account.inTransactionsCount.toLocaleString()}
-                </span>
-              </>
-            }
-          />
-        </Col>
-        <Col md="2">
-          <CardCell
-            title="Locked"
-            imgLink="/static/images/icon-m-filter.svg"
-            text={<Balance amount={account.locked} />}
-          />
-        </Col>
-        <Col md="4">
-          <CardCell
-            title="Storage (Used / Paid)"
-            imgLink="/static/images/icon-m-block.svg"
-            text={`${account.storageUsage.toLocaleString()} / ${account.storagePaidAt.toLocaleString()}`}
-          />
-        </Col>
-      </Row>
-      <Row noGutters className="border-0">
-        <Col md="4">
-          <CardCell
-            title="Created"
-            text={
-              typeof account.timestamp === "number"
-                ? moment(account.timestamp).format(
-                    "MMMM DD, YYYY [at] h:mm:ssa"
-                  )
-                : account.timestamp
-            }
-            className="block-card-created account-card-back border-0"
-          />
-        </Col>
-        <Col md="8">
-          <CardCell
-            title="Address"
-            text={account.address}
-            className="block-card-created-text account-card-back border-0"
-          />
-        </Col>
-      </Row>
-      <style jsx global>{`
-        .account-info-container {
-          border: solid 4px #e6e6e6;
-          border-radius: 4px;
-        }
+export default class extends React.Component<Props, State> {
+  state: State = {
+    storagePaidAtBlockHash: null
+  };
+
+  fetchBlockHash = async (height: string) => {
+    new BlocksApi()
+      .getBlockInfo(height)
+      .then(block => this.setState({ storagePaidAtBlockHash: block.hash }))
+      .catch(err => console.error(err));
+  };
+
+  componentDidMount() {
+    this.fetchBlockHash(this.props.account.storagePaidAt.toString());
+  }
```

Comment:
I think we also need to trigger it on `componentDidUpdate` as otherwise clicking on another account link while on Account Details page will lead to a problem of re-rendering with the `storagePaidAtBlockHash` from the previous account.